src/in_progress.rs

```diff
@@ -1110,3 +1110,132 @@ pub fn is_hardware_waiting_repair(
     let stmt_query = conn.query(&stmt, &params)?;
     Ok(!stmt_query.is_empty())
 }
+
+/// Get region id based on the region name.
+pub fn get_region_id(pool: &Pool<ConnectionManager>, region_name: &str) -> BynarResult<Option<u32>> {
+    let conn = get_connection_from_pool(pool)?;
+
+    // Get region Id from region name
+    let stmt = "SELECT region_id FROM regions WHERE region_name = $1";
+    let stmt_query = conn.query(stmt, &[&region_name])?;
+
+    let mut region_id: u32 = 0;
```

Comment:
You can remove this variable now since the function returns an Option<u32>.  So now all you need to do is return Ok(Some(region_id)) at 1129.

Reply:
completed